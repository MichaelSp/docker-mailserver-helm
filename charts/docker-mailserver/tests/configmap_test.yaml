suite: configmap
templates:
  - configmap.yaml
tests:

  - it: should correctly import configmaps from config/ if demo mode is disabled
    set:
      demoMode.enabled: false
    asserts:
      - matchRegex:
          path: data.fail2ban-fail2ban\.cf
          pattern: "dbpurgeage"        


  - it: should configure imaps port 10993 if proxyProtocol enabled
    set:
      proxyProtocol.enabled: true      
    asserts:
      - matchRegex:
          path: data.dovecot\.cf
          pattern: 10993

  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}

  - it: configures openDKIM
    set:
      openDKIM.TrustedHosts: something
    asserts:
      matchRegex:
        path: data.TrustedHosts
        pattern: something