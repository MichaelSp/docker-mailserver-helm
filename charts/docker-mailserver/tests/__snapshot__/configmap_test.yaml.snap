manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      dovecot.cf: "\nhaproxy_trusted_networks = 10.0.0.0/8 192.168.0.0/16 172.16.0.0/16\nservice imap-login {\n    inet_listener imap {\n        port = 143\n    }\n  \n    inet_listener imaps {\n        port = 993\n        ssl = yes\n    }\n  \n    inet_listener imap_proxy {\n        haproxy = yes\n        port = 10143\n        ssl = no\n    }\n\n    inet_listener imaps_proxy {\n        haproxy = yes\n        port = 10993\n        ssl = yes\n    }\n}\n"
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
        chart: docker-mailserver-0.1.0
        heritage: Helm
        release: RELEASE-NAME
      name: dovecot-cf
  2: |
    apiVersion: v1
    data:
      fts-xapian-plugin.conf: ""
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
        chart: docker-mailserver-0.1.0
        heritage: Helm
        release: RELEASE-NAME
      name: fts-xapian-plugin-conf
  3: |
    apiVersion: v1
    data:
      user-patches.sh: "#!/bin/bash\n# Make sure to keep this file in sync with https://github.com/docker-mailserver/docker-mailserver/blob/master/target/postfix/master.cf!\ncat <<EOS >> /etc/postfix/master.cf\n\n# Submission with proxy\n10587     inet  n       -       n       -       -       smtpd\n  -o syslog_name=postfix/submission\n  -o smtpd_tls_security_level=encrypt\n  -o smtpd_sasl_auth_enable=yes\n  -o smtpd_sasl_type=dovecot\n  -o smtpd_reject_unlisted_recipient=no\n  -o smtpd_sasl_authenticated_header=yes\n  -o smtpd_client_restrictions=permit_sasl_authenticated,reject\n  -o smtpd_relay_restrictions=permit_sasl_authenticated,reject\n  -o smtpd_sender_restrictions=\\$mua_sender_restrictions\n  -o smtpd_discard_ehlo_keywords=\n  -o milter_macro_daemon_name=ORIGINATING\n  -o cleanup_service_name=sender-cleanup\n  -o smtpd_upstream_proxy_protocol=haproxy  \n\n# Submissions with proxy\n10465     inet  n       -       n       -       -       smtpd\n  -o syslog_name=postfix/submissions\n  -o smtpd_tls_wrappermode=yes\n  -o smtpd_sasl_auth_enable=yes\n  -o smtpd_sasl_type=dovecot\n  -o smtpd_reject_unlisted_recipient=no\n  -o smtpd_sasl_authenticated_header=yes\n  -o smtpd_client_restrictions=permit_sasl_authenticated,reject\n  -o smtpd_relay_restrictions=permit_sasl_authenticated,reject\n  -o smtpd_sender_restrictions=\\$mua_sender_restrictions\n  -o smtpd_discard_ehlo_keywords=\n  -o milter_macro_daemon_name=ORIGINATING\n  -o cleanup_service_name=sender-cleanup\n  -o smtpd_upstream_proxy_protocol=haproxy\n\n# Smtp with proxy\n12525     inet  n       -       n       -       1       postscreen\n  -o syslog_name=postfix/smtp-proxy\n  -o postscreen_upstream_proxy_protocol=haproxy\n  -o postscreen_cache_map=btree:$data_directory/postscreen_10025_cache\nEOS\n"
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
        chart: docker-mailserver-0.1.0
        heritage: Helm
        release: RELEASE-NAME
      name: user-patches-sh
